### PR TITLE
test: fix test_acl_deluser

### PR DESCRIPTION
**Pull request by BorysTheDev**
the test contains 2 bugs:
1) wasn't set the key pattern
2) incorrect error type for EXEC command

### CODE REVIEW

File: tests/dragonfly/acl_family_test.py [hunk 1]
```diff
@@ -220,27 +220,23 @@ async def test_acl_cat_commands_multi_exec_squash(df_factory):
...
 
-    res = await client.execute_command("ACL SETUSER george ON >pass +@transaction +@string")
-    assert res == b"OK"
-
-    res = await client.execute_command("AUTH george pass")
-    assert res == b"OK"
+    assert await client.execute_command("ACL SETUSER george ON >pass +set ~*") == b"OK"
+    assert await client.execute_command("AUTH george pass") == b"OK"
 
-    await client.execute_command("MULTI")
```
Comment (kostasrim) [inline tests/dragonfly/acl_family_test.py:234]: This should `not` be removed. It was the whole point of the test, to `delete while we are in multi` and that is the reason it was failing. 

Plz bring `MULTI/EXEC` back.

**Comment by kostasrim** [comment]
and also plz for each test case you fix, write down a sort description that explains the reason that it was failing such that we reviewers have context + if we ever go back to this PR we don't have to rereview the code

File: tests/dragonfly/acl_family_test.py [hunk 2]
```diff
@@ -220,27 +220,31 @@ async def test_acl_cat_commands_multi_exec_squash(df_factory):
...
 async def test_acl_deluser(df_server):
     client = aioredis.Redis(port=df_server.port)
 
-    res = await client.execute_command("ACL SETUSER george ON >pass +@transaction +@string")
-    assert res == b"OK"
-
-    res = await client.execute_command("AUTH george pass")
-    assert res == b"OK"
+    assert (
+        await client.execute_command("ACL SETUSER george ON >pass +@transaction +set ~*") == b"OK"
```
Comment (kostasrim) [inline tests/dragonfly/acl_family_test.py:228]: Also not that I refactored the code slightly, `df_server.client()` above and remove the "b" it's noise :D

Comment (romange) [reply tests/dragonfly/acl_family_test.py:228]: if it currently passes, it's not a noise. The right fix is use `async_client` fixture that is already configured with `decode_responses`.

Comment (kostasrim) [reply tests/dragonfly/acl_family_test.py:228]: @romange read my comment. I said use `df_server.client()` which uses `decode_responses` internally. If he does that (or use async_client -- it doesn't matter) he can remove the `"b"` which is noise

Comment (kostasrim) [reply tests/dragonfly/acl_family_test.py:228]: And I also already pushed a refactor for this

File: tests/dragonfly/acl_family_test.py [hunk 3]
```diff
@@ -220,27 +220,31 @@ async def test_acl_cat_commands_multi_exec_squash(df_factory):
...
-    await client.execute_command("SET key 44")
+    assert await client.execute_command("MULTI") == b"OK"
+    assert await client.execute_command("SET the_answer 42") == b"QUEUED"
 
     admin_client = aioredis.Redis(port=df_server.port)
-    await admin_client.execute_command("ACL DELUSER george")
+    assert await admin_client.execute_command("ACL DELUSER george") == 1
 
-    with pytest.raises(redis.exceptions.ConnectionError):
+    # the connection was destroyed so EXEC will be executed in the new connection without MULTI
```
Comment (kostasrim) [inline tests/dragonfly/acl_family_test.py:236]: What do you mean here? The connection is not destroyed ? There are two different connections. One the `client` which is authed as `george` and one the admin which is authed as `default`

**Comment by kostasrim** [comment]
I pushed the changes to test this in a loop with reg tests

File: tests/dragonfly/acl_family_test.py [hunk 4]
```diff
@@ -220,27 +220,29 @@ async def test_acl_cat_commands_multi_exec_squash(df_factory):
...
+    with pytest.raises(redis.exceptions.ResponseError):
         await client.execute_command("EXEC")
 
-    await admin_client.close()
+    assert await client.execute_command("ACL WHOAMI") == "User is default"
+
+    with pytest.raises(redis.exceptions.AuthenticationError):
+        await client.execute_command("AUTH george pass")
+
```
Comment (kostasrim) [inline tests/dragonfly/acl_family_test.py:244]: also these are not really needed and covered by unit tests

File: tests/dragonfly/acl_family_test.py (hunk 3 above)
Comment (BorysTheDev) [reply tests/dragonfly/acl_family_test.py:236]: deluser destroy the connection

**Comment by kostasrim** [comment]
I have been trying to look back on why I disabled this test and not fix it (if it was just because the type error). I managed to find https://github.com/dragonflydb/dragonfly/pull/2057/files but I did not write any context :face_exhaling: . I also filtered the regression tests to look back on the failed runs at that time but the logs/history is not there because github deletes it :cry: 

I will run the reg tests 5 times, if it passes I don't want us to waste more time and just move forward. If it fails again we will cover it anyway and skipping it way less effort than rerunning this again and again or trying to remember what was the reason.

I am very very skeptical of why I wrote it fails on arm :face_exhaling:

File: .github/actions/regression-tests/action.yml [hunk 5]
```diff
@@ -37,7 +37,7 @@ runs:
         export DRAGONFLY_PATH="${GITHUB_WORKSPACE}/${{inputs.build-folder-name}}/${{inputs.dfly-executable}}"
         export UBSAN_OPTIONS=print_stacktrace=1:halt_on_error=1 # to crash on errors
 
-        timeout 50m pytest -m "${{inputs.filter}}" --durations=10 --color=yes --json-report --json-report-file=report.json dragonfly --log-cli-level=INFO || code=$?
+        timeout 50m pytest -m "${{inputs.filter}}" --durations=10 --color=yes --json-report --json-report-file=report.json dragonfly/acl_family_test.py --log-cli-level=INFO || code=$?
```
Comment (romange) [inline .github/actions/regression-tests/action.yml:40]: should it be reverted?

Comment (BorysTheDev) [reply .github/actions/regression-tests/action.yml:40]: yes, Kostas run it to check robustness

**Review by kostasrim — DISMISSED** [review]
Passed 3 times on triggered runs. Hope it remains like that :)

**Review by kostasrim — APPROVED** [review]

**Pull request merged**